server/protocols/fetchmail/src/test/java/org/apache/james/fetchmail/FetchMailTest.java

```diff
@@ -0,0 +1,68 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.fetchmail;
+
+import org.apache.commons.configuration2.BaseHierarchicalConfiguration;
+import org.apache.commons.configuration2.HierarchicalConfiguration;
+import org.apache.commons.configuration2.tree.ImmutableNode;
+import org.apache.james.core.Username;
+import org.apache.james.user.api.UsersRepository;
+import org.junit.jupiter.api.Test;
+import org.mockito.MockedConstruction;
+import org.mockito.Mockito;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.tuple;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class FetchMailTest {
+
+    @Test
+    void shouldAddAccountWithGoodDataFetchMail() throws Exception {
+        try (MockedConstruction<ParsedConfiguration> mockParsedConfiguration = Mockito.mockConstruction(ParsedConfiguration.class)) {
+            UsersRepository fieldLocalUsers = mock(UsersRepository.class);
+            when(fieldLocalUsers.countUsers()).thenReturn(4);
+            when(fieldLocalUsers.list()).thenReturn(List.of(Username.of("user23@domain.com")).iterator()).thenReturn(List.of(Username.of("user23@domain.com")).iterator());
```

Comment:
```suggestion
            when(fieldLocalUsers.list()).thenReturn(List.of(Username.of("user23@domain.com")).iterator());
```
think is enough :)

Reply:
Could we indent this?

Reply:
Thank you for the review. The thenReturn is added twice because it is called twice in the test. Otherwise the first iterator is returned on second call so it has already been iterated over and returns no value on hasNext.

I have indented the file using the formatting.xml mentioned https://james.apache.org/contribute.html. Am i missing something?

Reply:
What @chibenwa means by indentation is something looking like this:

```
when(fieldLocalUsers.list())
    .thenReturn(List.of(Username.of("user23@domain.com")).iterator())
    .thenReturn(List.of(Username.of("user23@domain.com")).iterator());
```

> The thenReturn is added twice because it is called twice in the test.

Ah maybe yes, those iterators can be tricky sometimes :) Fair enough!